lib/cc/cli/version_checker.rb

```diff
@@ -0,0 +1,83 @@
+require "uuid"
+
+module CC
+  module CLI
+    class VersionChecker
+      include CC::CLI::Output
+
+      VERSION_CHECK_TIMEOUT = 60 * 60 # 1 Hour in seconds
+      DEFAULT_VERSIONS_URL = "https://versions.codeclimate.com".freeze
+
+      def check
+        return unless check_version?
+
+        print_new_version_message if outdated?
+      end
+
+      private
+
+      def check_version?
+        first_arg = ARGV.first
+        if %w[--check-version].include? first_arg
+          ARGV.shift
+          true
+        else
+          false
+        end
+      end
+
+      def outdated?
+        api_response["outdated"] == true
+      end
+
+      def latest_version
+        api_response["latest"]
+      end
+
+      def print_new_version_message
+        warn "A new version (v#{latest_version}) is available"
+      end
+
+      def api_response
+        @api_response ||=
+          begin
+            JSON.parse(api_response_body)
+          rescue JSON::ParserError => error
+            CLI.debug(error)
+            # We don't know so pretend all is peachy. We'll try again next time.
```

Comment:
Generally I try hard to avoid Pokemon exception handling (catch all exceptions). This is an expected one and this class knows how to handle it. There are other exceptions though. For example, Ctrl-C is an exception and this class has no business handling it but it may be handled somewhere else.